### ISSUE TITLE
One-Hot-Encoding does not match the current number of countries in Geoguessr

### ISSUE THREAD

**Issue by felixbusch1** [jordanbehrendt/good_practices_ml#17]
The current Geoguessr dataset contains 124 folders. The file "utils/country_list/country_list.csv" does not contain exactly 124 entries. This may cause the One-Hot-Encoding to be inaccurate when applied to the dataset.

**Comment by lbrenig** [comment]
The country_list is based on the countries [CLIP was trained to predict](https://github.com/openai/CLIP/blob/main/data/country211.md).
The one hot encodings and the model output is based on that number of countries (211).
Unless I'm missing something, the one hot encodings and labels are always in reference to the country list:

https://github.com/jordanbehrendt/good_practices_ml/blob/1597a75b1b0ce593085395065a68ee8479cc24b4/finetuning/model/model_trainer.py#L247-L248

(Here the string labels are mapped to the country_list)

**Comment by labrunhosarodrigues** [comment]
So the issue is not the number matching, but the fact that the country list changes some of the names wrt. those in the original geoguesser data. While those changes are reported in the paper, we should have a script that automatically performs that transformation.

**Comment by lbrenig** [comment]
> So the issue is not the number matching, but the fact that the country list changes some of the names wrt. those in the original geoguesser data. While those changes are reported in the paper, we should have a script that automatically performs that transformation.

I believe this is being addressed in #18
